### PR TITLE
[Security] Add `statusCode` and `message` arguments to `#[IsGranted]`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Following https://github.com/symfony/symfony-docs/pull/18101#issuecomment-1476562770

### CODE REVIEW

**Comment by WebMamba** [comment]
Great thanks @alexandre-daubois! I was searching for it after looking at your last PR!

**Comment by MrYamous** [comment]
I am also tempted to explicitly specify the default status code, and maybe document `message` argument (or I can do another PR for that) WDYT ?

**Comment by alexandre-daubois** [comment]
Updated! What do you think?

File: security.rst [hunk 1]
```diff
@@ -2299,13 +2299,27 @@ the ``ROLE_SUPER_ADMIN`` permission:
...
 
+A custom status code can be set with the ``statusCode`` argument. This
+code will be used in the HTTP response::
+
+    // src/Controller/AdminController.php
+    // ...
+
+    use Symfony\Component\Security\Http\Attribute\IsGranted;
+
+    #[IsGranted('ROLE_ADMIN', statusCode: 403)]
```
Comment (MrYamous) [inline security.rst]: I think we can keep a custom statusCode here and specify in text above 403 is default value, seems confusing to me to explicitly define default value here

File: security.rst [hunk 2]
```diff
@@ -2299,13 +2299,27 @@ the ``ROLE_SUPER_ADMIN`` permission:
     #[IsGranted('ROLE_ADMIN')]
     class AdminController extends AbstractController
     {
-        #[IsGranted('ROLE_SUPER_ADMIN')]
+        #[IsGranted('ROLE_SUPER_ADMIN', message: 'You are not allowed to access the admin dashboard.')]
```
Comment (MrYamous) [inline security.rst]: We should maybe mention `message` is optional ?

File: security.rst (hunk 1 above)
Comment (alexandre-daubois) [reply security.rst]: You're right, updated!

File: security.rst (hunk 2 above)
Comment (alexandre-daubois) [reply security.rst]: Added a little comment above this line :+1:

File: security.rst [hunk 3]
```diff
@@ -2299,27 +2299,30 @@ the ``ROLE_SUPER_ADMIN`` permission:
...
     use Symfony\Component\Security\Http\Attribute\IsGranted;
 
-    #[IsGranted('ROLE_ADMIN', statusCode: 403)]
+    #[IsGranted('ROLE_ADMIN', statusCode: 423)]
     class AdminController extends AbstractController
     {
         // ...
     }
 
+The default code is 403.
```
Comment (OskarStark) [inline security.rst]:
```suggestion
```

already mentioned in the sentence above. Afterwards good to merge 👍

**Comment by OskarStark** [comment]
Thank you @alexandre-daubois.

**Pull request closed without merging**